### PR TITLE
Redux I

### PR DESCRIPTION
**Pull request by josepplloo**
## Summary

 - Redux finger exercise solved.

## Screenshots
react1

## Trello Card

https://trello.com/c/1rkTaSoF/13-redux-i-actions-reducers-store

### CODE REVIEW

File: react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js [hunk 1]
```diff
@@ -9,15 +9,26 @@ const initialState = {
...
+      return { ...state, books: action.payload };
     case actions.ADD_TO_CART: // TODO to implement the logic
-      return { ...state };
+      return { ...state, bookSelected: [...state.bookSelected, action.payload] };
     case actions.ADD_ITEM: // TODO to implement the logic
-      return { ...state };
+      return {
+        ...state,
+        bookSelected: state.bookSelected.map(item =>
+          item.id === action.payload ? { ...item, quantity: (item.quantity += 1) } : { ...item }
```
Comment (erikankodelia) [inline react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js]: Excellent! only one comment. It is not necessary return { ...item } when the condition is false:

item.id === action.payload ? { ...item, quantity: (item.quantity += 1) } : item

File: react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js [hunk 2]
```diff
@@ -9,15 +9,26 @@ const initialState = {
...
-      return { ...state };
+      return {
+        ...state,
+        bookSelected: state.bookSelected.filter(item => item.id !== action.payload)
+      };
     case actions.SEARCH_ITEM: // TODO to implement the logic
-      return { ...state };
+      return {
+        ...state,
+        books: state.books.filter(book => book.name.toLowerCase().indexOf(action.payload.toLowerCase()) > -1)
```
Comment (erikankodelia) [inline react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js]: You can use the method 'includes' like this:

book.name.toLowerCase().includes(action.payload.toLowerCase())

File: react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js [hunk 3]
```diff
@@ -9,15 +9,39 @@ const initialState = {
...
-      return { ...state };
-    case actions.SEARCH_ITEM: // TODO to implement the logic
-      return { ...state };
+      return {
+        ...state,
+        bookSelected: state.bookSelected.filter(item => item.id !== action.payload)
+      };
+    case actions.SEARCH_ITEM: {
+      // TODO to implement the logic
+      const originalData = state.originalData.length === 0 ? state.books : state.originalData;
```
Comment (cvega08) [inline react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js]: ```state.originalData.length ? state.originalData : state.books;```

File: react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js [hunk 4]
```diff
@@ -9,15 +9,39 @@ const initialState = {
...
+      const filterText = action.payload.toLowerCase().trim();
+      let filterBooks = [];
+
+      if (filterText !== '') {
```
Comment (cvega08) [inline react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js]: `if ( filterText ) { ... }`

File: react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js [hunk 5]
```diff
@@ -9,15 +9,39 @@ const initialState = {
...
-      return { ...state };
-    case actions.SEARCH_ITEM: // TODO to implement the logic
-      return { ...state };
+      return {
+        ...state,
+        bookSelected: state.bookSelected.filter(item => item.id !== action.payload)
+      };
+    case actions.SEARCH_ITEM: {
+      // TODO to implement the logic
+      const originalData = state.originalData.length ? state.originalData : state.books;
```
Comment (erikankodelia) [inline react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js]: I think that when you get the books, you can save it in "originalData" too. So you can you state.originalData directly.

File: react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js [hunk 6]
```diff
@@ -9,15 +9,39 @@ const initialState = {
...
+      const filterText = action.payload.toLowerCase().trim();
+      let filterBooks = [];
+
+      if (filterText) {
```
Comment (erikankodelia) [inline react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js]: It is perfect! but if you want,  you can write it like this:

const books = filterText ? state.originalData.filter(book => book.name.toLowerCase().includes(filterText)) : state.originalData

File: react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js [hunk 7]
```diff
@@ -9,15 +9,39 @@ const initialState = {
...
+        filterBooks = originalData.filter(book => book.name.toLowerCase().includes(filterText));
+      } else {
+        filterBooks = state.originalData;
+      }
+
+      return {
+        ...state,
+        originalData,
```
Comment (erikankodelia) [inline react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js]: And here it is not necessary to update it

**Pull request merged**